### DISCUSSION TITLE
AutoUSDZ broken in iOS 17.5 beta

### DISCUSSION THREAD

**Discussion by milesgreen** [google/model-viewer#4741] [General]
With the new iOS 17.5 beta release (21F5048f) there is a regression in behaviour when opening a USDZ blob to AR QuickLook from Safari.

In iOS 17.5 beta, when opening a USDZ blob, the user is presented with a prompt that asks:

> Do you want to download "model.usdz"?

With options to "View" or "Download"

Selecting "Download", downloads the file (not ideal).

Selecting "View" displays the default USDZ thumbnail image, with an AR icon at top right. The user must tap again to open the model in AR. 

Closing the AR QuickLook view returns the user to the USDZ thumbnail preview. From here the user must use the back button to get back to the originating HTML page.

This seems like a Safari/iOS issue. Bug filed here:
https://bugs.webkit.org/show_bug.cgi?id=272321

<img width="785" alt="auto-usdz-error" src="https://github.com/google/model-viewer/assets/53870482/07abf825-b158-4770-951c-117480eb3d66">

**Comment by milesgreen** [comment]
@elalish - Is there anyone from Apple we should be tagging in these threads?

**Reply by elalish** [reply]
I used to tag @grorg, but he never responded, except once to say he didn't work on quick look anymore. I'm not sure if there's anyone now.

**Reply by toji** [reply]
@zachernuk and @AdaRoseCannon are the people I would reach out to initially. They're not going to be the relevant engineers but I'm hopeful that they will be able to figure out where to route the request and, more importantly, be able to advocate for the web-facing impact of these changes.

**Reply by AdaRoseCannon** [reply]
The fastest way to get it resolved is to report it via feedback assistant. Once you have reported it let me know the number and I can do my best to keep track of it.

**Reply by milesgreen** [reply]
Thanks @AdaRoseCannon - I'll do that.

**Comment by milesgreen** [comment]
Tracking more information over here: https://github.com/google/model-viewer/issues/4746

**Comment by milesgreen** [comment]
This is now resolved in iOS 17.5 beta 4.
